### PR TITLE
Restructure code to get rid of optional dependencies

### PR DESCRIPTION
**Pull request by amitmalav**
This PR contains changes which are required to run `broker v2 apis` only with minimal settings.
Hence Unnecessary dependencies on modules like (cf, fabrik) were delegated.
Dependency on `fabrik` is moved from `FabrikBaseController` to individual controllers which need fabrik module.

Sample minimal settings file can be found [here](https://gist.github.com/amitmalav/452a32d9c7c2c3fea2a32f5e7de744dc)

### CODE REVIEW

File: broker/lib/index.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 exports.middleware = require('./middleware');
-exports.fabrik = require('./fabrik');
 exports.bootstrap = bootstrap;
 const config = require('../../common/config');
 if (config.enable_swarm_manager) {
   exports.docker = require('../../data-access-layer/docker');
 }
 const logger = require('../../common/logger');
 const docker = exports.docker;
+require('./fabrik/UnlockResourcePoller');
```
Comment (subhankarc) [inline broker/lib/index.js]: move this UnlockResourcePoller out of fabrik.

Comment (subhankarc) [reply broker/lib/index.js]: put it in common.

File: common/utils/index.js [hunk 2]
```diff
@@ -590,4 +592,17 @@ function deploymentStaggered(err) {
   const response = _.get(err, 'error', {});
   const description = _.get(response, 'description', '');
   return description.indexOf(CONST.FABRIK_OPERATION_STAGGERED) > 0 && description.indexOf(CONST.FABRIK_OPERATION_COUNT_EXCEEDED) > 0;
+}
+
+function getPlatformManager(context) {
```
Comment (subhankarc) [inline common/utils/index.js]: Use it in DirectorService as well.

**Comment by subhankarc** [comment]
@kamath-prasad  Can you please review and merge this? We want to merge it to master as this does not have any feature change, only dependency reduction.

File: broker/lib/index.js (hunk 1 above)
Comment (amitmalav) [reply broker/lib/index.js]: Done!

File: common/utils/index.js (hunk 2 above)
Comment (amitmalav) [reply common/utils/index.js]: Done!

**Pull request merged**